overlord/snapstate/snapstate.go

```diff
@@ -447,3 +447,39 @@ func GadgetInfo(s *state.State) (*snap.Info, error) {
 
 	return nil, state.ErrNoState
 }
+
+// MigrateToTypeInState migrates the state to have a snap type in the snap state of each setup snap.
+func MigrateToTypeInState(s *state.State, commit func()) error {
+	var stateMap map[string]*SnapState
+
+	s.Lock()
+	err := s.Get("snaps", &stateMap)
```

Comment:
Is there a race here? Can something else modify `snaps` while this runs (e.g. an install or something)? I.e. do we need to keep the lock for the entire function?